registry/storage/driver/gcs/gcs.go

```diff
@@ -369,17 +380,16 @@ type writer struct {
 	sessionURI string
 	buffer     []byte
 	buffSize   int
+	GCS        *storage.Client
 }
 
 // Cancel removes any written content from this FileWriter.
-func (w *writer) Cancel() error {
+func (w *writer) Cancel(ctx context.Context) error {
 	w.closed = true
-	err := storageDeleteObject(cloud.NewContext(dummyProjectID, w.client), w.bucket, w.name)
+	err := storageDeleteObject(context.Background(), w.bucket, w.name, w.GCS)
```

Comment:
Is there a reason why we're not using the `ctx` passed it via func param but instead creating a new `context` object?

Reply:
No. If I'm not mistaken receiving the context on the function is the one change I had to make to fix builds after rebasing this go around. I just didn't finish the thought  and update this. Should be done now.